_studio/mfx_lib/encode/mjpeg/src/mfx_mjpeg_encode.cpp

```diff
@@ -1102,13 +1104,12 @@ mfxStatus MFXVideoENCODEMJPEG::Init(mfxVideoParam *par_in)
     mfxStatus st = MFX_ERR_NONE, QueryStatus = MFX_ERR_NONE;
     mfxVideoParam* par = par_in;
 
-    if(m_isInitialized)
-        return MFX_ERR_UNDEFINED_BEHAVIOR;
+    MFX_CHECK(m_isInitialized, MFX_ERR_UNDEFINED_BEHAVIOR);
+    MFX_CHECK(par == NULL, MFX_ERR_NULL_PTR);
 
-    if(par == NULL)
-        return MFX_ERR_NULL_PTR;
+    MFX_CHECK(mfxInit() == ippStsNoErr, MFX_ERR_UNSUPPORTED);
```

Comment:
Don't do that, please. MFX_CHECK macro might (actually will) expand in a way that it will use expressions few times. In this case it will call mfxInit() few times which is not what we want. When you use MFX_CHECK like macro defs, always use trivial expressions, not functions. Should be something like:
```
auto ippst = mfxInit();
MFX_CHECK(ippst == ippStsNoErr, MFX_ERR_UNSUPPORTED);
```

Reply:
@dvrogozh , macro with multiple evaluation of some argument is time bomb. But in [the code](https://github.com/Intel-Media-SDK/MediaSDK/blob/c92b652bb0f9bb8504c54076a3a1db9fed3fa50e/_studio/shared/include/mfx_utils.h#L58) I see
`#define MFX_CHECK(EXPR, ERR)    { if (!(EXPR)) MFX_RETURN(ERR); }
`
So, it looks like EXPR is used only once.

It's matter of taste, and I agree that a code with potential side effects looks suspiciously in macro arguments.